src/raftstore/store/peer.rs

```diff
@@ -672,64 +567,115 @@ impl Peer {
             });
         }
 
-        ReadyResult {
-            apply_snap_result: apply_snap_result,
-            exec_results: vec![],
+        if apply_snap_result.is_some() {
+            worker.schedule(ApplyTask::register(self.peer_id(),
+                                              self.term(),
+                                              self.region_id,
+                                              self.get_store().apply_state.clone(),
+                                              self.get_store().applied_index_term,
+                                              self.region().clone()))
+                .unwrap();
         }
+
+        apply_snap_result
     }
 
-    pub fn handle_raft_ready_apply(&mut self, mut ready: Ready, result: &mut ReadyResult) {
+    pub fn handle_raft_ready_apply(&mut self, mut ready: Ready, worker: &mut Worker<ApplyTask>) {
         // Call `handle_raft_commit_entries` directly here may lead to inconsistency.
         // In some cases, there will be some pending committed entries when applying a
         // snapshot. If we call `handle_raft_commit_entries` directly, these updates
         // will be written to disk. Because we apply snapshot asynchronously, so these
         // updates will soon be removed. But the soft state of raft is still be updated
         // in memory. Hence when handle ready next time, these updates won't be included
         // in `ready.committed_entries` again, which will lead to inconsistency.
-        result.exec_results = if self.is_applying() {
+        if self.is_applying() {
             if let Some(ref mut hs) = ready.hs {
                 // Snapshot's metadata has been applied.
                 hs.set_commit(self.get_store().truncated_index());
             }
-            vec![]
         } else {
-            self.handle_raft_commit_entries(&ready.committed_entries)
+            let committed_entries = ready.committed_entries.take().unwrap();
+            for entry in &committed_entries {
+                self.raft_log_size_hint += entry.get_data().len() as u64;
+            }
+            committed_entries.iter()
+                .rev()
+                .any(|e| self.maybe_update_lease(e.get_term(), e.get_data()));
```

Comment:
Is it just to update the lease by getting the last entry propose time?

To be precise, it should process the lease updating in sequence for all entries, since the lease maybe extend in the sequence of entries committing.  
If it is to modify the lease updating to an imprecise behavior on purpose, due to the performance issue caused by the deserialization of all entries, maybe a comment could be used here to make it clear.